Zend/zend_inheritance.c

```diff
@@ -594,38 +628,78 @@ static ZEND_COLD zend_string *zend_get_function_declaration(const zend_function
-
-						if (Z_TYPE_P(zv) == IS_FALSE) {
-							smart_str_appends(&str, "false");
-						} else if (Z_TYPE_P(zv) == IS_TRUE) {
-							smart_str_appends(&str, "true");
-						} else if (Z_TYPE_P(zv) == IS_NULL) {
-							smart_str_appends(&str, "NULL");
-						} else if (Z_TYPE_P(zv) == IS_STRING) {
-							smart_str_appendc(&str, '\'');
-							smart_str_appendl(&str, Z_STRVAL_P(zv), MIN(Z_STRLEN_P(zv), 10));
-							if (Z_STRLEN_P(zv) > 10) {
-								smart_str_appends(&str, "...");
+						append_parameter_default_value(&str, zv);
+					}
+				} else {
+					zend_internal_arg_info *internal_arg_info = (zend_internal_arg_info*) arg_info;
+					if (internal_arg_info->default_value) {
+						zval code_zv;
+						zend_bool original_in_compilation;
+						uint32_t original_compiler_options;
+						zend_lex_state original_lex_state;
+						zend_ast *ast;
+						zend_arena *ast_arena;
+
+						size_t str1_length = strlen("<?php ");
+						size_t str2_length = strlen(internal_arg_info->default_value);
+						size_t str3_length = strlen(";");
+						size_t len = str1_length + str2_length + str3_length;
+						zend_string *code = zend_string_alloc(len, 0);
+
+						memcpy(ZSTR_VAL(code), "<?php ", str1_length);
+						memcpy(ZSTR_VAL(code) + str1_length, internal_arg_info->default_value, str2_length);
+						memcpy(ZSTR_VAL(code) + str1_length + str2_length, ";", str3_length);
+						ZSTR_VAL(code)[len] = '\0';
+						ZVAL_STR_COPY(&code_zv, code);
+
+						original_in_compilation = CG(in_compilation);
+						original_compiler_options = CG(compiler_options);
+						zend_save_lexical_state(&original_lex_state);
+
+						CG(in_compilation) = 1;
+						CG(compiler_options) |= ZEND_COMPILE_NO_CONSTANT_SUBSTITUTION | ZEND_COMPILE_NO_PERSISTENT_CONSTANT_SUBSTITUTION;
+
+						if (zend_prepare_string_for_scanning(&code_zv, "") == SUCCESS) {
+							CG(ast) = NULL;
+							CG(ast_arena) = zend_arena_create(1024 * 32);
+							LANG_SCNG(yy_state) = yycINITIAL;
+
+							if (zendparse() != 0) {
+								zend_ast_destroy(CG(ast));
+								zend_arena_destroy(CG(ast_arena));
+								CG(ast) = NULL;
 							}
-							smart_str_appendc(&str, '\'');
-						} else if (Z_TYPE_P(zv) == IS_ARRAY) {
-							smart_str_appends(&str, "Array");
-						} else if (Z_TYPE_P(zv) == IS_CONSTANT_AST) {
-							zend_ast *ast = Z_ASTVAL_P(zv);
-							if (ast->kind == ZEND_AST_CONSTANT) {
-								smart_str_append(&str, zend_ast_get_constant_name(ast));
+						}
+
+						ast = CG(ast);
+						ast_arena = CG(ast_arena);
+
+						if (ast) {
+							zend_ast_list *statement_list = zend_ast_get_list(ast);
+							zend_ast *const_expression_ast = statement_list->child[0];
+							if (const_expression_ast) {
+								zval zv;
+								zend_const_expr_to_zval(&zv, const_expression_ast);
+								append_parameter_default_value(&str, &zv);
+								zval_dtor(&zv);
 							} else {
-								smart_str_appends(&str, "<expression>");
+								smart_str_appends(&str, "<default>");
```

Comment:
I like this `<default>` text, but feel free to suggest a better one (should it be just `null` as before?)